config/logs.js

```diff
@@ -66,29 +66,52 @@ module.exports = {
       }
 
       // Sanitize OAuth client secrets
+      //
+      // This will display "00000...00000" in ELK.
       if (typeof metadata.request.query.client_secret === 'string') {
-        // display first/last three characters but scrub the rest
-        const sanitizedSecret = `${metadata.request.query.client_secret.slice(0, 3)}...${metadata.request.query.client_secret.slice(-3)}`;
+        // display first/last five characters but scrub the rest
+        const sanitizedSecret = `${metadata.request.query.client_secret.slice(0, 5)}...${metadata.request.query.client_secret.slice(-5)}`;
         metadata.request.query.client_secret = sanitizedSecret;
       }
 
+      // Sanitize OAuth client secrets found in Headers
+      //
+      // This will display "Basic 00000...00000" in ELK.
+      if (
+        typeof metadata.request.headers.authorization === 'string'
+        && metadata.request.headers.authorization.indexOf('Basic') !== -1
+      ) {
+        const sanitizedSecret = `${metadata.request.headers.authorization.slice(0, 11)}...${metadata.request.headers.authorization.slice(-5)}`;
+        metadata.request.headers.authorization = sanitizedSecret;
+      }
+
       // Sanitize JWTs, which can allow anyone to masquerade as this user.
       // We do this by removing the signature at the end, which only our
       // server can generate. That way, we can decode and inspect the payload
       // for debugging purposes, without risking the use of the JWT by devs
       // who have access to ELK.
-      if (typeof metadata.request.headers.authorization === 'string') {
+      if (
+        typeof metadata.request.headers.authorization === 'string'
+        && metadata.request.headers.authorization.indexOf('Bearer') !== -1
+      ) {
         let sanitizedJWT = metadata.request.headers.authorization.split('.');
-        const buffer = Buffer.from(sanitizedJWT[1], 'base64');
-        const asciiJWT = buffer.toString('ascii');
-        sanitizedJWT.pop();
-        sanitizedJWT = sanitizedJWT.join('.');
-        metadata.request.headers.authorization = sanitizedJWT;
-
-        // Auto-populate requesting user ID from JWT unless user.id was explicitly
-        // passed into the log.
-        if (typeof metadata.user.id === 'undefined') {
-          metadata.user.id = JSON.parse(asciiJWT).id;
+        if (sanitizedJWT.length === 3) {
```

Comment:
finally, this conditional was necessary to prevent HID failures during Drupal-based ["Extra secure auth process" step 2](https://github.com/UN-OCHA/hid_api/wiki/Integrating-with-HID-via-OAuth#step-2--request-access-and-id-tokens), where a non-JWT is being sent as a short-lived API key.